Output AST generated a syntax error

I get this error:
```
[ERROR] Formatter 'stylua' error: error: could not format from stdin: failed to format from stdin: INTERNAL ERROR: Output AST generated a syntax error. Please report this at https://github.com/johnnymorganz/stylua/issues
  error occurred while creating ast: unexpected token `console`
 ```
  When attempting to format this [example file](https://sourceforge.net/p/iup/iup/HEAD/tree/trunk/iup/html/examples/Lua/console.lua) from the IUP gui library.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Added flag `--no-ignore-vcs` to continue formatting files listed in a `.gitignore` file, instead of skipping over them ([#895](https://github.com/JohnnyMorganz/StyLua/issues/895))
 
+### Fixed
+
+- Fixed syntax error in output when a single-line comment appears between an index suffix and a table call argument, e.g. `foo.bar -- comment { }` ([#873](https://github.com/JohnnyMorganz/StyLua/issues/873))
+
 ## [2.3.1] - 2025-11-01
 
 ### Fixed

--- a/src/formatters/functions.rs
+++ b/src/formatters/functions.rs
@@ -1141,10 +1141,18 @@ pub fn format_function_call(
 
         let mut suffix = format_suffix(ctx, suffix, current_shape, ambiguous_next_suffix);
 
+        // Check if the previous suffix has trailing single-line comments.
+        // If it does, we MUST hang the AnonymousCall to avoid the opening parenthesis being commented out
+        let previous_suffix_has_trailing_singleline_comment = formatted_suffixes
+            .last()
+            .map(|s: &Suffix| s.has_trailing_comments(CommentSearch::Single))
+            .unwrap_or(false);
+
         // Hang the call, but don't hang if the previous suffix was an index and this is an anonymous call, i.e. `.foo()`
         if will_hang
             && !(previous_suffix_was_index
-                && matches!(suffix, Suffix::Call(Call::AnonymousCall(_))))
+                && matches!(suffix, Suffix::Call(Call::AnonymousCall(_)))
+                && !previous_suffix_has_trailing_singleline_comment)
         {
             suffix = trivia_util::prepend_newline_indent(ctx, &suffix, current_shape);
         }

--- a/tests/inputs/hang-call-chain-comments-4.lua
+++ b/tests/inputs/hang-call-chain-comments-4.lua
@@ -1,0 +1,15 @@
+-- https://github.com/JohnnyMorganz/StyLua/issues/873
+
+x = a.b -- comment
+{
+  y
+}
+
+console.dialog = iup.dialog
+{
+  iup.hbox -- use it to inherit margins
+  {
+    console.prompt,
+  },
+  title = "Command:",
+}

